### ISSUE TITLE
[sklearn#34647] OPTICS always triggers and EfficiencyWarning

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#303]
**Mirrored from [scikit-learn#34647](https://github.com/scikit-learn/scikit-learn/issues/34647)**

Original title: OPTICS always triggers and EfficiencyWarning
Original labels: Bug
Created: 2026-08-04T15:22:26Z

---

### Describe the bug and give evidence about its user-facing impact

Similar to https://github.com/scikit-learn/scikit-learn/issues/31030 but bug is in `OPTICS`

### Steps/Code to Reproduce

```python
import numpy as np
import sklearn
from sklearn.cluster import OPTICS
from sklearn.neighbors import kneighbors_graph, sort_graph_by_row_values

X = np.random.RandomState(0).randn(1000, 5)

graph = kneighbors_graph(X, n_neighbors=5, mode="distance")

OPTICS(metric="precomputed", min_samples=5).fit(graph)

graph=sort_graph_by_row_values(graph, warn_when_not_sorted=False)
OPTICS(metric="precomputed", min_samples=5).fit(graph)
```

### Expected Results

<img width="231" height="133" alt="Image" src="https://github.com/user-attachments/assets/1ffcceed-d251-43ae-94e4-b21c75c5f1c7" />

### Actual Results

In both cases:

```
Users/dealeon/projects/scikit-learn/sklearn/neighbors/_base.py:241:
EfficiencyWarning: Precomputed sparse input was not sorted by row values. Use the function sklearn.neighbors.sort_graph_by_row_values to sort the input by row values, with warn_when_not_sorted=False to remove this warning.
  warnings.warn(
```

<img width="193" height="117" alt="Image" src="https://github.com/user-attachments/assets/1da0a37a-9569-4a87-9218-458129c37382" />

### Versions

```shell
System:
    python: 3.13.9 | packaged by conda-forge | (main, Oct 22 2025, 23:38:18) [Clang 19.1.7 ]
executable: /Users/dealeon/mambaforge/envs/sklearn-dev/bin/python
   machine: macOS-26.5.2-arm64-arm-64bit-Mach-O

Python dependencies:
      sklearn: 1.10.dev0
          pip: 25.3
   setuptools: 80.9.0
        numpy: 2.3.4
        scipy: 1.16.3
       Cython: 3.2.0
       pandas: 2.3.3
   matplotlib: 3.10.7
       joblib: 1.5.2
threadpoolctl: 3.6.0
     narwhals: 2.11.0

Built with OpenMP: True

threadpoolctl info:
       user_api: blas
   internal_api: openblas
    num_threads: 8
         prefix: libopenblas
       filepath: /Users/dealeon/mambaforge/envs/sklearn-dev/lib/libopenblas.0.dylib
        version: 0.3.30
threading_layer: openmp
   architecture: VORTEX

       user_api: openmp
   internal_api: openmp
    num_threads: 8
         prefix: libomp
       filepath: /Users/dealeon/mambaforge/envs/sklearn-dev/lib/libomp.dylib
        version: None
```

### Interest in fixing the bug

Yes. I would like to open a PR for this issue.

It probably needs the same fix than this PR: https://github.com/scikit-learn/scikit-learn/pull/34399

EDIT: 
I opened this issue before a PR, thinking that I was following the rules of contributing to scikit-learn. 
- I forgot that some people forget to read (I described my interest on opening a PR).
- TIL that I should open an issue only if I need help. 
  

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Addressed in upstream PR(s). See scikit-learn/scikit-learn PRs for #34647.
